devito/passes/iet/mpi.py

```diff
@@ -368,3 +339,55 @@ def mpiize(graph, **kwargs):
         make_halo_exchanges(graph, mpimode=mpimode, **kwargs)
 
     make_reductions(graph, mpimode=mpimode, **kwargs)
+
+
+# *** Utilities
+
+def _filter_iter_mapper(iet):
+    """
+    Given an IET, return a mapper from Iterations to the HaloSpots.
+    Additionally, filter out Iterations that are not of interest.
+    """
+    iter_mapper = MapNodes(Iteration, HaloSpot, 'immediate').visit(iet)
+    iter_mapper = {k: [hs for hs in v if not hs.halo_scheme.is_void]
+                   for k, v in iter_mapper.items()}
+    iter_mapper = {k: v for k, v in iter_mapper.items() if k is not None}
+    iter_mapper = {k: v for k, v in iter_mapper.items() if len(v) > 1}
+
+    return iter_mapper
+
+
+def _make_cond_mapper(iet):
+    cond_mapper = MapHaloSpots().visit(iet)
+    return {hs: {i for i in v if i.is_Conditional and
```

Comment:
nitpicking (for code homogeneity), the `if` part of a multi-line comprehension goes on the next line, ie

```
[a for a in X
 if (condA and
     condB and
     condC)]
```

Reply:
actually, this is a nested comprehension, you might actually use an explicit for loop...